client/src/connection_cache.rs

```diff
@@ -1,32 +1,45 @@
 use {
     quinn::Endpoint,
     solana_connection_cache::{
-        client_connection::ClientConnection as BlockingClientConnection,
+        client_connection::ClientConnection,
         connection_cache::{
-            ConnectionCache as BackendConnectionCache, NewConnectionConfig, ProtocolType,
+            BaseClientConnection, ConnectionCache as BackendConnectionCache, ConnectionPool,
+            NewConnectionConfig,
         },
-        nonblocking::client_connection::ClientConnection as NonblockingClientConnection,
     },
-    solana_quic_client::{QuicConfig, QuicConnectionManager},
-    solana_sdk::{pubkey::Pubkey, signature::Keypair},
+    solana_quic_client::{QuicConfig, QuicConnectionManager, QuicPool},
+    solana_sdk::{pubkey::Pubkey, signature::Keypair, transport::Result as TransportResult},
     solana_streamer::streamer::StakedNodes,
-    solana_udp_client::UdpConnectionManager,
+    solana_udp_client::{UdpConfig, UdpConnectionManager, UdpPool},
     std::{
         error::Error,
         net::{IpAddr, Ipv4Addr, SocketAddr},
         sync::{Arc, RwLock},
     },
 };
 
-pub const DEFAULT_CONNECTION_POOL_SIZE: usize = 4;
-pub const DEFAULT_CONNECTION_CACHE_USE_QUIC: bool = true;
-pub const MAX_CONNECTIONS: usize = 1024;
```

Comment:
This is another change to the public api. Please restore, and consider marking with deprecated tags.

Reply:
The first 2 were only added in https://github.com/solana-labs/solana/pull/29774 and are not in a stable release yet.
`MAX_CONNECTIONS` is not public in `v1.14` either:
https://github.com/solana-labs/solana/blob/v1.14/client/src/connection_cache.rs#L30-L40

I think it should be fine to update the api before stable release, no?

(also these shouldn't have been public to begin with and should have been caught in original reviews).

Reply:
https://docs.rs/solana-client/1.14.14/solana_client/connection_cache/index.html
But looks like the breakage predates you. And now that v1.15.0 has been cut, I guess the ship has sailed.
I don't disagree with you that they should not have been public to begin with, but I am trying to get us to uphold our backward compatibility policy as best we can.